benchbuild/utils/actions.py

```diff
@@ -259,17 +182,18 @@ def __init__(self, project: "benchbuild.project.Project") -> None:
     def __str__(self, indent: int = 0) -> str:
         return textwrap.indent("* Execute configured action.", indent * " ")
 
-    def __call__(self) -> StepResult:
-        raise NotImplementedError
-
     def onerror(self) -> None:
         Clean(self.project)()
 
+    @abc.abstractmethod
+    def __call__(self) -> StepResult:
+        raise NotImplementedError
 
-StepTy = tp.TypeVar("StepTy", bound=Step)
 
+StepTy_co = tp.TypeVar("StepTy_co", bound=Step, covariant=True)
```

Comment:
You want this still bound on `Step`?

Ok, from my side I have two "issues" here.
a) Protocols can only inherit protocols, so not `Step`
b) I change the call function

b -> I can fix by calling my function a different name

Reply:
No not necessarily.
This seems like a minimal patch to improve type-safety in utils/actions.py.

Removing the bound requires more changes by all MultiStep descendants:
- Any
- Experiment
- RequireAll

I haven't actually thought about implementing them there yet, so that might be a minimal change as well.